runtime/src/main/java/io/quarkiverse/fury/FuryProducer.java

```diff
@@ -3,22 +3,23 @@
 import jakarta.enterprise.inject.Produces;
 import jakarta.inject.Singleton;
 
+import org.apache.fury.BaseFury;
 import org.apache.fury.Fury;
 
 /**
  * Producers of beans that are injectable via CDI.
  */
 @Singleton
 public class FuryProducer {
-    private volatile Fury fury;
+    private volatile BaseFury fury;
```

Comment:
Hmm, just wonder what is the difference between `Fury` and `BaseFury`? why we have to expose `BaseFury` to the users?

Reply:
OK, I see. The `BaseFury` is a common interface. How about changing `BaseFury` to `Fury` and let `Fury` with `DefaultFury`? I think it might be much clearer.

Reply:
We'd better  not do that, Fury is used across the whole code repository, it's a  huge change.